client/components/ProductOverview/ThumbnailView.jsx

```diff
@@ -0,0 +1,76 @@
+import React, { useState, useEffect } from 'react';
+import { AiOutlineArrowDown, AiOutlineArrowUp } from 'react-icons/ai';
+import PropTypes from 'prop-types';
+
+const ThumbnailView = ({ photos, handleClick, currentIndex }) => {
+  const [thumbnailStart, setThumbnailStart] = useState(currentIndex);
+  const [thumbnailEnd, setThumbnailEnd] = useState(7);
+
+  useEffect(() => {
+    if (photos.length <= 7) {
+      setThumbnailStart(0);
+      setThumbnailEnd(photos.length);
+    } else if (photos.length > 7) {
+      if (currentIndex > photos.length - 7) {
+        setThumbnailStart(photos.length - 7);
+        setThumbnailEnd(photos.length);
+      } else if (currentIndex < 7) {
+        setThumbnailStart(0);
+        setThumbnailEnd(7);
+      }
+    }
+  }, [currentIndex]);
+
+  const handleArrowUp = () => {
+    if (currentIndex > thumbnailStart && currentIndex <= thumbnailEnd - 1) {
+      handleClick(currentIndex - 1);
+    } else if (currentIndex === thumbnailStart) {
+      setThumbnailStart(thumbnailStart - 1);
+      setThumbnailEnd(thumbnailEnd - 1);
+      handleClick(currentIndex - 1);
+    }
+  };
+
+  const handleArrowDown = () => {
+    if (currentIndex >= thumbnailStart && currentIndex < thumbnailEnd - 1) {
+      handleClick(currentIndex + 1);
+    } else if (currentIndex >= thumbnailEnd - 1) {
+      setThumbnailStart(thumbnailStart + 1);
+      setThumbnailEnd(thumbnailEnd + 1);
+      handleClick(currentIndex + 1);
+    }
+  };
+
+  return (
+    <div className="arrowsThumbnailContainer">
+      {currentIndex !== 0 ? <AiOutlineArrowUp className="upArrow" size={45} onClick={handleArrowUp} /> : null}
```

Comment:
I would make the size prop more extensible by using a variable instead of a magic number out of thin air. It would definitely make it more readable and you'd only have to change it in once place in the future if anything changes.

Reply:
added arrowsize variable